tests/kubernetes/test_client.py

```diff
@@ -78,3 +80,23 @@ def test_disable_verify_ssl(self):
         else:
             configuration = Configuration()
         assert not configuration.verify_ssl
+
+    def test_api_client_retry_configuration_default(self):
+        conf = initialize_config()
+        api_client_retry_configuration = conf.getjson(
+            "kubernetes", "api_client_retry_configuration", fallback={}
+        )
+
+        assert api_client_retry_configuration == {}
+
+    @mock.patch("airflow.kubernetes.kube_client.conf")
+    @conf_vars({("kubernetes", "api_client_retry_configuration"): '{"total": 3, "backoff_factor": 0.5}'})
+    def test_api_client_retry_configuration_correct_values(self, conf):
+        get_kube_client(in_cluster=False)
+        conf.getboolean.assert_called_with("kubernetes", "api_client_retry_configuration")
+
+        api_client_retry_configuration = conf.getjson(
+            "kubernetes", "api_client_retry_configuration", fallback={}
+        )
+
+        assert api_client_retry_configuration == {"total": 3, "backoff_factor": 0.5}
```

Comment:
You can replace it by this test:
```suggestion
    @mock.patch("kubernetes.config.incluster_config.InClusterConfigLoader")
    @conf_vars({("kubernetes", "api_client_retry_configuration"): '{"total": 3, "backoff_factor": 0.5}'})
    def test_api_client_retry_configuration_correct_values(self, mock_in_cluster_loader):
        get_kube_client(in_cluster=True)
        client_configuration = mock_in_cluster_loader().load_and_set.call_args.args[0]
        assert client_configuration.retries.total == 3
        assert client_configuration.retries.backoff_factor == 0.5
```

Reply:
Added the changes, thanks.
However, I would like to understand what `@mock.patch("kubernetes.config.incluster_config.InClusterConfigLoader")` does @hussein-awala

Reply:
This line patches the class `InClusterConfigLoader` (which is used to load k8s config when in_cluster is True) and replaces it by a mock.
You can check the mock [documentation](https://docs.python.org/3/library/unittest.mock.html#the-patchers) for more info.

Reply:
Thank you. Was not aware of this. Thanks!